src/components/network/geo-data.js

```diff
@@ -88,4 +90,67 @@ export default class GeoData {
 
         return [ substationPosition1, substationPosition2 ];
     };
+
+    getLineDistances(positions) {
+        if (positions !== null && positions.length > 1) {
+            let cumulativeDistanceArray = [0];
+            let cumulativeDistance = 0;
+            let segmentDistance;
+            let ruler;
+            for (let i = 0; i < positions.length - 1; i++) {
+                ruler = cheapRuler(positions[i][1], 'meters');
+                segmentDistance = ruler.lineDistance(positions.slice(i, i + 2));
+                cumulativeDistance = cumulativeDistance + segmentDistance;
+                cumulativeDistanceArray[i+1] = cumulativeDistance;
+            }
+            return cumulativeDistanceArray;
+        }
+        return null;
+    }
+
+    getCoordinateInLine(positions, cumulativeDistances, percent) {
+        if (percent > 100 || percent < 0) {
+            throw new Error("percent value incorrect: " + percent);
+        }
+        if (cumulativeDistances === null || cumulativeDistances.length < 2 || cumulativeDistances[1] === 0) {
```

Comment:
Why the ``` cumulativeDistances[1] === 0``` check ?

Reply:
added a comment to explain it : "checks if the line is composed of only one segment with a distance = 0 (a few line are like that in some files)" which kind of means nothing and breaks the algo

Reply:
what if the line has more than 1 segment, but the first 2 points have the same coordinates ?
what if cumulativeDistance has length 1 ?

maybe instead check for cumulativeDistance[length -1] == 0 ?

Reply:
Indeed, changed.